pkg/infrastructure/azure/azure.go

```diff
@@ -357,6 +363,26 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 		}
 	}
 
+	if installConfig.Azure.CloudName == aztypes.StackCloud {
+		imageClientOpts := p.clientOptions
+		imageClientOpts.APIVersion = "2020-06-01"
```

Comment:
Should we clone the entire client option so that the APIVersion set does not impact elsewhere (i.e. p.clientOptions is a pointer)? Otherwise, we should reset the APIVersion after this block?

```go
imageClientOpts := p.clientOptions.Clone()
```

Reply:
Oops I missed this comment when working through the feedback. Good catch.

I do the same override in the PostProvision hook, so I solved this problem by creating another set of options `computeClientOptions`, so now that field is used here and in the post provision hook.

I squashed the commits, because I'm trying to keep this merge ready.

Reply:
Ahh okay! That makes sense. Thanks!